### DISCUSSION TITLE
Doesn't automatically check new version available

### DISCUSSION THREAD

**Discussion by yzenati** [zimfw/zimfw#503] [Q&A] [answered]
Hello !

First, thank you for this project !
I was very disappointed by other framework, zimfw is so light and easy to configure/custom.

I am on macOS and I noticed that I have never seen a mention of a new version available as it is documented (every 30 days) and I didn't disable it.

Is it a normal behavior ?

Thanks !

**Comment by ericbn** [comment]
Hi @yzenati.

What's the output of the `cat ${ZIM_HOME}/.latest_version` command in your terminal?

Note that this check for a new version only applies to new versions of the `zimfw` plugin manager, not new versions of installed modules. The latest version of `zimfw` currently is `1.11.3`.

**Comment by yzenati** [comment, marked as answer]
Ok, I look through the code, and I think that I misunderstood how this option works.
I thought it would run the same way as oh-my-zsh and display the message at terminal startup, but it seems it only prints the message when the command `zimfw` is called, am I right ?

Is there any way that the check could happen at shell startup ? How about trigger the `_zimfw_version_check`or an equivalent inside the `.zim/init.zsh` ?

Last question, do you think it could be conceivable to find a way to also check if modules need to update too ? Is it technically verifiable using something like `_zimfw_run_tool` `update` way ?

What about writing a module to trigger those behaviors ?

**Reply by ericbn** [reply]
You're correct in your observations about the current behavior.

I can add an action to zimfw to check if a new version of zimfw is available. Checking if updates for the modules are available should be possible too. In case of git I guess we'll have to do `fetch` and then check if there are new commits in the specified or default branch.

**Reply by yzenati** [reply]
Oh yeah, that would be so great to have those features !

Thanks for your response !

**Reply by ericbn** [reply]
Released version 1.12.0 with a new `zimfw check-version` action that checks if a new version of zimfw is available, and a new `zimfw check` action that checks if updates for modules are available.

**Reply by yzenati** [reply]
Awesome ! Thank you very much !
